### DISCUSSION TITLE
How do I use enter to submit a form?

### DISCUSSION THREAD

**Discussion by zenhob** [storybookjs/storybook#26666] [Help] [answered]
### Summary

I have a React component with a form and a single text input. When "enter" is pressed in the text input, the form is submitted, triggering the submit callback. This component works fine in an application context. However, in a storybook story, hitting "enter" to submit does not do anything at all.

The only issue I could find that seemed related to this was extremely old (https://github.com/storybookjs/storybook/issues/128) and I am hoping the state of the art might have improved here. **How do I get my component to behave the same way in Storybook as it does in my application?**

### Additional information

```
$ npx storybook --version
7.6.17
$ cat .storybook/main.js 
module.exports = {
  docs: {
    autodocs: true,
  },
  stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],
  addons: [
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    '@storybook/addon-interactions',
    'storybook-addon-apollo-client',
  ],
  framework: {
    name: '@storybook/nextjs',
    options: {
      nextConfigPath: '../next.config.js',
    },
  },
};

```

**Comment by zenhob** [comment]
I forgot to add: i can add a submit button to the form, and click it in the storybook! What I CANNOT do is submit by hitting "enter" from a textbox, no matter what else is in the form.

**Comment by zenhob** [comment]
I'm leaving this open, because I want to understand why this is happening, but for anyone else: I worked around this by adding an `onKeyDown` handler that listens for the enter key.

**Comment by Marklb** [comment, marked as answer]
Could you provide a reproduction? I tried creating a story, with a form and a text input as the only child, but pressing enter seems to trigger the submit.

If it is simple to reproduce then you can quickly make a reproduction with: https://storybook.new/

**Comment by zenhob** [comment]
Thank you Mark, that's kind of you to take the time. I'm perfectly ready to think it's something in else in the stack that I don't realize is interfering. Sorry for troubling you.
